Feature/cgd 35

Dan-Y-Ko opened the pull request:
Only added banner to ideation page. Other pages should be the same for the most part, however the layout needs to be changed a bit to make it look good with this banner. That will be taken care of in another pr after the side nav is finished.

Removed docker and backend related things.

marktlinn reviewed (commented):
The code looks good to me. I don't know if it's only on my machine, but the image for the banner section looks a little squashed and maybe a little blury.

Dan-Y-Ko commented:
> The code looks good to me. I don't know if it's only on my machine, but the image for the banner section looks a little squashed and maybe a little blury.

Could you show me what it looks like? I don't think any of us have been too concerned about how it looks on different screen sizes and I haven't been addressing yet because I would like the side nav finished so we have more to work with regarding how the components are laid out and screen sizes.

marktlinn commented:
> > The code looks good to me. I don't know if it's only on my machine, but the image for the banner section looks a little squashed and maybe a little blury.
> 
> Could you show me what it looks like? I don't think any of us have been too concerned about how it looks on different screen sizes and I haven't been addressing yet because I would like the side nav finished so we have more to work with regarding how the components are laid out and screen sizes. I'm not sure about the blurry thing though (maybe related)?


OK, no worries, I wasn't testing it at a small break point, just fullscreen on a laptop, the image just looked like it wasn't very good quality, but I don't think it's a big deal at this stage, as it can always be changed out for a higher quality version later.

Dan-Y-Ko commented:
Ok so I added quite a bit to this pr.

Summary:

- Added eslint rule to disable relative imports from a parent directory.
- added pr template
- re-organized the barrel exports

To explain how the barrel exports work:

- Each folder will have its own barrel export file, named index.ts
- The export in a barrel must be a named export.
- When importing from same directory, prefer using the barrel file, unless there's a circular dependency (it is an issue in the store folder), this can potentially clean up the list of imports quite a lot.
- When adding a subfolder to a parent's barrel file, just use export * from that folder. See root components folder for example.
- When importing from a parent directory, always use absolute import and to the barrel file.

marktlinn reviewed (approved):
LGTM

Dan-Y-Ko commented on cypress/screenshots/2-advanced-examples/misc.cy.js/my-image.png:
no...not sure where that came from

Dan-Y-Ko replied:
I just did my typical code changes and push lol
